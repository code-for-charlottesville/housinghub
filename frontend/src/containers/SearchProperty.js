import React from "react";
import { connect } from "react-redux";
import "../style/App.css";
import { Container, Row, Col } from "react-bootstrap";
import SearchFiltersSidebar from "../components/SearchFiltersSidebar";
import SearchTable from "../components/SearchTable";

const SearchProperty = (props) => {
  let columns = [
    {
      field: "address",
      title: "Address",
    },
    {
      field: "property_owner",
      title: "Property Owner",
    },
    {
      field: "monthly_rent",
      title: "Rent",
    },
    {
      field: "voucher_type_accepted",
      title: "Vouchers Accepted",
    },
  ];

  //Populate rows
  let rows = [];
  for (let i = 0; i < props.searchResults.results.length; i++) {
    let r = [
      props.searchResults.results[i].address,
      props.searchResults.results[i].property_name,
      props.searchResults.results[i].monthly_rent,
      props.searchResults.results[i].bedrooms,
      props.searchResults.results[i].bathrooms,
      <a href="">See More</a>,
    ];
    rows.push(r);
  }

  rows = [["Apartment", "520 E Main Street", "$2500", 5, 1, "Test"]];

  return (
    <Container className="search-container">
      <Row>
        <Col className="search-filters-container">
          <SearchFiltersSidebar />
        </Col>
        <Col xs={9} className="search-table-container">
          <SearchTable rows={rows} columns={columns} />
        </Col>
      </Row>
    </Container>
  );
};

function mapStateToProps(state) {
  return {
    searchResults: state.search.searchResults,
  };
}

export default connect(mapStateToProps)(SearchProperty);
