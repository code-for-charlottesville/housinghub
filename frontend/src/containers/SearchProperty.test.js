import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import sampleMockStore from "../../test/mockStore.js";
import SearchProperty from "./SearchProperty";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";

const mockStore = configureStore([]);

describe("containers/SearchProperty", () => {
  let _matchesSnapshot = (C, props = {}, path) => {
    let store = mockStore({
      ...sampleMockStore,
      search: {
        ...sampleMockStore.search,
        searchResults: props.searchResults,
      },
    });
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={[path]}>
          <Provider store={store}>
            <C {...props} />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  };

  let testTable = [
    {
      name: "redirects to login if not logged in",
      component: SearchProperty,
      path: "/property/search",
      props: {
        searchResults: {
          pagination: {
            page: 0,
            results_per_page: 0,
            totalNumberOfResults: 0,
          },
          results: [
            {
              address: "string",
              allow_criminal_records: true,
              application_fee: 0,
              background_screening_company: "string",
              bathrooms: 0,
              bedrooms: 0,
              bus_line: true,
              contact_method: ["string"],
              credit_screening_company: "string",
              date_first_available: "string",
              deposit: 0,
              elevator: true,
              floor: 0,
              has_basement: true,
              housing_type: "string",
              id: "string",
              is_available: true,
              landlord_id: "string",
              last_contact_date: "string",
              last_contacted_by: "string",
              last_month_rent_required: true,
              listing_date: "string",
              monthly_rent: 0,
              navigator_id: "string",
              potential_month_available: 0,
              property_name: "string",
              school_district: "string",
              shared_bathrooms: 0,
              unit_apt_no: "string",
              voucher_type_accepted: ["string"],
              voucher_type_not_accepted: ["string"],
              wheelchair_accessibility: true,
              where_listed: ["string"],
              year_available: 0,
              zip_code: "string",
            },
          ],
        },
      },
    },
  ];

  testTable.forEach((t) => {
    it(t.name, () => {
      _matchesSnapshot(t.component, t.props, t.path);
    });
  });
});
