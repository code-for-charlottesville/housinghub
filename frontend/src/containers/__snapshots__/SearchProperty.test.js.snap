// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/SearchProperty redirects to login if not logged in 1`] = `
<div>
  <div
    className="search-page row"
  >
    <div
      className="pr-0 border col-xl-2 col-lg-3 col-12"
    >
      <div
        className="size border"
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": "50px",
            }
          }
        >
          <h5
            style={
              Object {
                "color": "grey",
                "fontFamily": "Roboto",
                "fontSize": "16px",
                "fontWeight": "bold",
                "paddingLeft": "12px",
                "paddingTop": "50px",
              }
            }
          >
            Search Fields
          </h5>
        </div>
        <hr
          style={
            Object {
              "borderWidth": "3px",
            }
          }
        />
        <div>
          <p
            style={
              Object {
                "fontFamily": "Roboto",
                "fontSize": "14px",
                "marginBottom": "5px",
                "paddingLeft": "15px",
              }
            }
          >
            Rent per unit
          </p>
          <form
            className=""
          >
            <div
              className="form-row"
              style={
                Object {
                  "marginBottom": "5px",
                  "paddingLeft": "15px",
                  "paddingRight": "15px",
                }
              }
            >
              <div
                className="col"
              >
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Max"
                />
              </div>
              <div
                className="col"
              >
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Max"
                />
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col"
              >
                <p
                  style={
                    Object {
                      "fontFamily": "Roboto",
                      "fontSize": "13px",
                      "paddingLeft": "20px",
                    }
                  }
                >
                  Min
                </p>
              </div>
              <div
                className="col"
              >
                <p
                  style={
                    Object {
                      "fontFamily": "Roboto",
                      "fontSize": "13px",
                      "paddingLeft": "0px",
                    }
                  }
                >
                  Max
                </p>
              </div>
            </div>
            <div
              className="form-row"
              style={
                Object {
                  "marginTop": "17px",
                  "paddingLeft": "15px",
                  "paddingRight": "15px",
                }
              }
            >
              <div
                className="col-12"
              >
                <label
                  className="form-label"
                >
                  Date Available
                </label>
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="MM-dd-YYYY"
                  type="date"
                />
              </div>
            </div>
            <div
              className="form-group"
              style={
                Object {
                  "marginTop": "24px",
                  "paddingLeft": "9px",
                  "paddingRight": "9px",
                }
              }
            >
              <label
                className="form-label"
                htmlFor="bedrooms"
              >
                Bedrooms
              </label>
              <select
                className="custom-select"
                id="numBeds"
                onChange={[Function]}
                value={4}
              >
                <option>
                  1
                </option>
                <option>
                  2
                </option>
                <option>
                  3
                </option>
                <option>
                  4
                </option>
                <option>
                  5
                </option>
                <option>
                  6
                </option>
                <option>
                  7
                </option>
              </select>
            </div>
            <div
              className="form-group"
              style={
                Object {
                  "marginTop": "15px",
                  "paddingLeft": "9px",
                  "paddingRight": "9px",
                }
              }
            >
              <label
                className="form-label"
                htmlFor="bathrooms"
              >
                Bathrooms
              </label>
              <select
                className="custom-select"
                id="numBath"
                onChange={[Function]}
                value={3.5}
              >
                <option>
                  1
                </option>
                <option>
                  1.5
                </option>
                <option>
                  2
                </option>
                <option>
                  2.5
                </option>
                <option>
                  3
                </option>
                <option>
                  3.5
                </option>
                <option>
                  4
                </option>
                <option>
                  4.5
                </option>
                <option>
                  5
                </option>
              </select>
            </div>
            <div
              className="form-row"
              style={
                Object {
                  "marginTop": "20px",
                }
              }
            >
              <div
                className="col-10"
                style={
                  Object {
                    "paddingLeft": "15px",
                  }
                }
              >
                Deposit Needed
              </div>
              <div
                className="col-1"
              >
                <div
                  className="form-check"
                >
                  <input
                    aria-label="deposit"
                    className="form-check-input position-static"
                    disabled={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                </div>
              </div>
            </div>
            <div
              className="form-row"
              style={
                Object {
                  "marginTop": "4px",
                }
              }
            >
              <div
                className="col-10"
                style={
                  Object {
                    "paddingLeft": "15px",
                  }
                }
              >
                Credit Report Needed
              </div>
              <div
                className="col-1"
              >
                <div
                  className="form-check"
                >
                  <input
                    aria-label="creditReport"
                    className="form-check-input position-static"
                    disabled={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                </div>
              </div>
            </div>
            <div
              className="form-row"
              style={
                Object {
                  "marginTop": "4px",
                }
              }
            >
              <div
                className="col-10"
                style={
                  Object {
                    "paddingLeft": "15px",
                  }
                }
              >
                Near Bus Stop
              </div>
              <div
                className="col-1"
              >
                <div
                  className="form-check"
                >
                  <input
                    aria-label="bus"
                    className="form-check-input position-static"
                    disabled={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                </div>
              </div>
            </div>
            <div
              className="form-row"
              style={
                Object {
                  "marginBottom": "30px",
                  "marginTop": "4px",
                }
              }
            >
              <div
                className="col-10"
                style={
                  Object {
                    "paddingLeft": "15px",
                  }
                }
              >
                Pets allowed
              </div>
              <div
                className="col-1"
              >
                <div
                  className="form-check"
                >
                  <input
                    aria-label="pets"
                    className="form-check-input position-static"
                    disabled={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                </div>
              </div>
            </div>
            <hr
              style={
                Object {
                  "borderWidth": "2px",
                  "margin": "15px",
                }
              }
            />
            <div
              className="form-group"
              style={
                Object {
                  "marginTop": "15px",
                  "paddingLeft": "9px",
                  "paddingRight": "9px",
                }
              }
            >
              <label
                className="form-label"
                htmlFor="schoolDistrict"
              >
                School District
              </label>
              <select
                className="custom-select"
                id="schoolDistrict"
                onChange={[Function]}
              >
                <option>
                  Alemarle
                </option>
                <option>
                  City of Charlottesville
                </option>
                <option>
                  University of Virginia
                </option>
              </select>
            </div>
            <hr
              style={
                Object {
                  "borderWidth": "2px",
                  "margin": "15px",
                }
              }
            />
            <div
              className="form-group"
              style={
                Object {
                  "marginTop": "15px",
                  "paddingLeft": "9px",
                  "paddingRight": "9px",
                }
              }
            >
              <label
                className="form-label"
                htmlFor="leaseType"
              >
                Lease Type
              </label>
              <select
                className="custom-select"
                id="leaseType"
              >
                <option>
                  Month to Month
                </option>
                <option>
                  2
                </option>
                <option>
                  3
                </option>
                <option>
                  4
                </option>
                <option>
                  5
                </option>
              </select>
            </div>
            <hr
              style={
                Object {
                  "borderWidth": "2px",
                  "margin": "15px",
                }
              }
            />
            <div
              className="form-group"
              style={
                Object {
                  "marginTop": "15px",
                  "paddingLeft": "9px",
                  "paddingRight": "9px",
                }
              }
            >
              <label
                className="form-label"
                htmlFor="vouchers"
              >
                Vouchers Accepted
              </label>
              <select
                className="custom-select"
                id="vouchers"
                onChange={[Function]}
              >
                <option>
                  Any
                </option>
                <option>
                  2
                </option>
                <option>
                  3
                </option>
                <option>
                  4
                </option>
                <option>
                  5
                </option>
              </select>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      className="pl-0 search-table col-xl-10 col-lg-9 col-12"
    >
      
      <div
        className="border mb-4"
      >
        <table
          className="table table-hover"
        >
          <thead
            className="bg-info text-white"
          >
            <tr>
              <th>
                Property Type
              </th>
              <th>
                Address
              </th>
              <th>
                Rent
              </th>
              <th>
                Bedrooms
              </th>
              <th>
                Bathrooms
              </th>
              <th>
                Delete
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className=""
              onClick={[Function]}
            >
              <td>
                string
              </td>
              <td>
                string
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <button
                className="btn btn-primary"
                disabled={false}
                type="button"
              >
                Delete
              </button>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="text-center pagination ml-3 row"
      >
        <div
          className="col-sm-1"
        />
        <div
          className="col-sm-9"
        >
          <div
            className="btn-group"
            role="group"
          >
            <button
              className="PaginationBox btn btn-secondary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <a>
                Next
              </a>
            </button>
            <button
              className="PaginationBox btn btn-secondary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <a>
                Last
              </a>
            </button>
          </div>
        </div>
        <div
          className="col-sm-2"
        >
          <button
            className="justify-end btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Add Property
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
